system/View/View.php

```diff
@@ -269,7 +269,7 @@ public function render(string $view, array $options = null, bool $saveData = nul
 
 		$this->logPerformance($this->renderVars['start'], microtime(true), $this->renderVars['view']);
 
-		if ($this->debug && (! isset($options['debug']) || $options['debug'] === true))
+		if (($this->debug && (! isset($options['debug']) || $options['debug'] === true)) && in_array('CodeIgniter\Filters\DebugToolbar', service('filters')->getFiltersClass()['after']))
```

Comment:
can use strict in_array check: 

```suggestion
              if (($this->debug && (! isset($options['debug']) || $options['debug'] === true)) && in_array('CodeIgniter\Filters\DebugToolbar', service('filters')->getFiltersClass()['after'], true))
```